build-system/compile/sources.js

```diff
@@ -151,8 +151,11 @@ const THIRD_PARTY_TRANSFORM_GLOBS = [
   'third_party/optimized-svg-icons/social-share-svgs.js',
 ];
 
+const SERVER_TRANSFORM_PATH = 'build-system/server/new-server/transforms';
```

Comment:
Similar comment here. This constant has more to do with the server transforms than the innards of the compilation toolchain. Can you move it to the (newly renamed) `build-system/server/typescript-compile.js` and export it from there instead?